### PR TITLE
Add initial spec to confirm presence of Black King

### PR DESCRIPTION
**Pull request by radiofreemattd**
So the chief thing to understand here is a bit of a weird cycle (not that weird, but still)...

The `populate_board!` method gets called upon the creation of a `Game`, which means that our test needs to create a `Game` and because a game is owned by a `User`, we need those as well. Once we have created all the dependencies, we can create the game, which fires the method we are testing.

I am not sure I have chosen the best way to test that the method is correctly working (and I generally don't like more than one expectation per test), so I am completely open to other approaches here.

Hopefully, someone can more or less copy this for the remaining pieces...

### CODE REVIEW

**Comment by jonmast** [comment]
This spec should be in the model spec not the controller spec. The spec itself looks great :+1: One thing I would change is to use FactoryGirl for creating the game. It should automatically create the players for you as well.

**Comment by mscoutermarsh** [comment]
Agree, should be a model spec :smile:

**Comment by radiofreemattd** [comment]
I think this also has the benefit of making sure that everything, other than that which is being tested, is stubbed out. Good catch on not using the games factory...I'm glad I did this as I am getting a *much* better understanding of factories...

**Comment by jonmast** [comment]
:+1: Looks great :tada:

File: spec/models/game_spec.rb [hunk 1]
```diff
@@ -2,12 +2,22 @@
...
   it { is_expected.to belong_to :black_player }
 
   it { is_expected.to have_many :chess_pieces }
 
-  it { is_expected.to validate_presence_of :name }
+  describe '#create' do
+    context 'when initializing the state of the board' do
+      it 'puts a Black King in the correct starting position' do
+        game = create :game
+        black_king = game.chess_pieces.where(position_x: 4, position_y: 0).first
```
Comment (mscoutermarsh) [inline spec/models/game_spec.rb]: an alternative syntax for this would be

`game.chess_pieces.find_by(position_x: 4, position_y: 0)`

**Pull request merged**